### PR TITLE
Создание проекта

### PR DESCRIPTION
**Pull request by NikitaGoroshko7**
https://trello.com/c/XTJkOOg1/20-%D1%81%D0%BE%D0%B7%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5-%D0%BF%D1%80%D0%BE%D0%B5%D0%BA%D1%82%D0%B0
Created ProjectController
Added actions for ProjectController
Added Views
Added general css file and added scripts
Added migration and created table CreateProject in the database
Added UploadedFileService and changed in the ProfileController
Added folder for storage ProjectImages

### CODE REVIEW

File: CrowdFundingApplication/Models/CreateProject.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+public class CreateProject
+{
+    public Guid Id { get; set; }
+    public string NameOfProject { get; set; }
+    public string ImageName { get; set; }
+    public string Description { get; set; }
+    public string Category { get; set; }
+    public int Sum { get; set; }
+    public string Date { get; set; }
```
Comment (GalayM) [inline CrowdFundingApplication/Models/CreateProject.cs]: string -> DateTime

File: CrowdFundingApplication/Models/CreateProject.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿namespace CrowdFundingApplication.Models;
+
+public class CreateProject
+{
+    public Guid Id { get; set; }
+    public string NameOfProject { get; set; }
+    public string ImageName { get; set; }
+    public string Description { get; set; }
+    public string Category { get; set; }
+    public int Sum { get; set; }
```
Comment (GalayM) [inline CrowdFundingApplication/Models/CreateProject.cs]: int? Not decimal?

File: CrowdFundingApplication/Models/CreateProject.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿namespace CrowdFundingApplication.Models;
+
+public class CreateProject
+{
+    public Guid Id { get; set; }
+    public string NameOfProject { get; set; }
+    public string ImageName { get; set; }
+    public string Description { get; set; }
+    public string Category { get; set; }
```
Comment (GalayM) [inline CrowdFundingApplication/Models/CreateProject.cs]: string -> enum

File: CrowdFundingApplication/Models/CreateProjectViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
+﻿namespace CrowdFundingApplication.Models;
+
+public class CreateProjectViewModel
```
Comment (GalayM) [inline CrowdFundingApplication/Models/CreateProjectViewModel.cs]: It is View or Edit model? If View then there should be no validation.

File: CrowdFundingApplication/Models/CreateProjectViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    public Category Category { get; set; }
+
+    [Required(ErrorMessage = "Не указана сумма проекта!")]
+    public int Sum { get; set; }
+
+    [Required(ErrorMessage = "Не указана дата окончания проекта!")]
+    public string Date { get; set; }
+}
+
+public enum Category
```
Comment (GalayM) [inline CrowdFundingApplication/Models/CreateProjectViewModel.cs]: Each entity in a separate file

File: CrowdFundingApplication/Views/Project/Index.cshtml [hunk 6]
```diff
@@ -0,0 +1,80 @@
+﻿@{
+    ViewData["Title"] = "Правила создания проекта";
+}
+
+<form method="post" asp-action="Index">
+<br />
```
Comment (GalayM) [inline CrowdFundingApplication/Views/Project/Index.cshtml]: Remove br

File: CrowdFundingApplication/Views/Project/Index.cshtml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                                    <div class="form-check">
+                                      <input class="form-check-input" type="checkbox" name="check" id="checkbox" onclick="change_button(this, 'button')">
+                                      <label class="label_name form-check-label" for="defaultCheck1">Я принимаю положение об использовании личной (персональной) информации при размещении проектов на платформе ViewStar и 
+                                          соглашаюсь с условиями использования сервиса народного финансирования (краудфандинг)</label>
+                                    </div>
+                                </div>
+                                
+                                <div class="col-lg-3 mt-4 mb-5">
+                                    <button class="sub_button btn btn-primary btn-block" disabled id="button" type="submit">Создать проект</button>
+                                    <br />
```
Comment (GalayM) [inline CrowdFundingApplication/Views/Project/Index.cshtml]: Remove all BRs

File: CrowdFundingApplication/Controllers/ProjectController.cs [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+                ModelState.AddModelError("Image", "Ошибка! Для создание проекта необходима обложка!");
+            }
+            else
+            {
+                model.Date = DateTime.Parse(model.Date).ToShortDateString();
+                model.ImageName = UploadedFileService.UploadedFile(model.Image, _webHostEnvironment, "ProjectImages");
+
+                var obj = GetCreateProject(model);
+                _db.Add(obj);
+                _db.SaveChanges();
```
Comment (GalayM) [inline CrowdFundingApplication/Controllers/ProjectController.cs]: It is better to use asynchronous implementation.

File: CrowdFundingApplication/Models/CreateProjectViewModel.cs (hunk 4 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Models/CreateProjectViewModel.cs]: It is edit model

File: CrowdFundingApplication/Controllers/ProjectController.cs (hunk 8 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Controllers/ProjectController.cs]: Has been changed

File: CrowdFundingApplication/Views/Project/Index.cshtml (hunk 7 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Views/Project/Index.cshtml]: Has been changed

File: CrowdFundingApplication/Views/Project/Index.cshtml (hunk 6 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Views/Project/Index.cshtml]: Has been changed

File: CrowdFundingApplication/Models/CreateProjectViewModel.cs (hunk 5 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Models/CreateProjectViewModel.cs]: Has been changed

File: CrowdFundingApplication/Models/CreateProject.cs (hunk 2 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Models/CreateProject.cs]: Has been changed

File: CrowdFundingApplication/Models/CreateProject.cs (hunk 1 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Models/CreateProject.cs]: Has been changed

File: CrowdFundingApplication/Models/CreateProject.cs (hunk 3 above)
Comment (NikitaGoroshko7) [reply CrowdFundingApplication/Models/CreateProject.cs]: Has been changed

File: CrowdFundingApplication/Models/CreateProjectViewModel.cs [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    public string Description { get; set; }
+
+    [Required(ErrorMessage = "Не выбрана категория проекта!")]
+    public Category Category { get; set; }
+
+    [Required(ErrorMessage = "Не указана сумма проекта!")]
+    public int Sum { get; set; }
+
+    [Required(ErrorMessage = "Не указана дата окончания проекта!")]
+    public string Date { get; set; }
```
Comment (GalayM) [inline CrowdFundingApplication/Models/CreateProjectViewModel.cs]: string -> DateTime
int -> decimal

File: CrowdFundingApplication/Controllers/ProjectController.cs [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    public IActionResult CreateProjectInDetails() => View();
+
+    [HttpGet]
+    public IActionResult Organization() => View();
+
+    private CreateProject GetCreateProject(EditProjectViewModel model)//convert CreateProjectViewModel in CreateProject for record in the database
+    {
+        var project = new CreateProject
+        {
+            Id = Guid.NewGuid(),
```
Comment (GalayM) [inline CrowdFundingApplication/Controllers/ProjectController.cs]: You do not need to fill in the ID, they must be generated by themselves

File: CrowdFundingApplication/Models/Category.cs [hunk 11]
```diff
@@ -0,0 +1,41 @@
+﻿namespace CrowdFundingApplication.Models;
+
+public enum Category
```
Comment (GalayM) [inline CrowdFundingApplication/Models/Category.cs]: Category -> CategoryEnum

File: CrowdFundingApplication/wwwroot/js/site.js [hunk 12]
```diff
@@ -1,4 +1,27 @@
...
-// Write your JavaScript code.
+    if (checkbox.checked == true) {
+        btn.disabled = "";
+    } else {
+        btn.disabled = "disabled";
+    }
+}
+
+function showCount2() {
+    resultid.innerHTML = 125;
```
Comment (GalayM) [inline CrowdFundingApplication/wwwroot/js/site.js]: move 125 to viable

**Review by GalayM — APPROVED** [review]

**Pull request merged**